sound/soc/sof/ipc4-topology.c

```diff
@@ -976,7 +993,7 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		}
 	}
 
```

Comment:
Now as we supports multiple output pin and sink pin, we can't use the index of 'pin_fmt' as the output pin index.
So I think the below code 
```
	if (out_format) {
		memcpy(out_format, &available_fmt->source_audio_fmts[i].audio_fmt,
		       sizeof(*out_format));
		dev_dbg(sdev->dev, "Init output audio formats for %s\n", swidget->widget->name);
		sof_ipc4_dbg_audio_format(sdev->dev, &available_fmt->source_audio_fmts[i], 1);
	}
```
needs refine.

Let's imagine the below scenarios: 
1. There are 2 sink pins and 1 output pins and support 16KHz and 48KHz. We will totally get 4 sink fmts and 2 output fmts. The output fmt can't be 1:1 mapped to the sink fmt, unless we do some tricks when generating the tplgs to make their index 1:1 mapping. 
2. There are 1 sink pin and 2 output pins. How can we get output format for both output pins?

As sink pins format and output pins format are in pairs. Maybe we need an index in the format structure. When we match the pin0 sink format, we can easily get all other pins format based on the format index.

Reply:
@libinyang audio formats dont have to be in pairs but they should be in the pin order 0 -> N. Maybe I should spell out that the number of intput/output formats for pin 0 must match. and should go first.

For non-zero pins, atm we pick the first available format. This needs to be revisited later

Reply:
> @libinyang audio formats dont have to be in pairs but they should be in the pin order 0 -> N. Maybe I should spell out that the number of intput/output formats for pin 0 must match. and should go first.
> 
> For non-zero pins, atm we pick the first available format. This needs to be revisited later

Let's image module A supports: If sink_pin0 and sink_pin1 is 48KHz, its output is 48KHz. If sink_pin0 and sink_pin1 is 16KHz, its output is 16KHz. Then we will have 2 audio format set:
format set1: sink_pin0 48KHz, sink_pin1 48KHz, output_pin0 48KHz.
format set2: sink_pin0 16KHz, sink_pin1 16KHz, output_pin0 16KHz.

Should we support such scenario? Or we only support sink_pin1 always be set 48KHz in this case?

Reply:
@libinyang in this case we will limit to only one set in tplg based on what this module is connected to.

Reply:
@ranj063 why would the number of input / output formats need to match for pin0? It's clearly wrong for a SRC component that produces a fixed-frequency result.

Reply:
@plbossart i dont mean input/output formats match as int their values. What I mean is for every input params, there's a set of output params available for pin 0